### PR TITLE
add externalTx integration tests using cardano-transactions

### PR DESCRIPTION
**Pull request by paweljakubas**
# Issue Number
#2041 

# Overview

- [x] Single-output external tx (for cardano-transactions construction) for Shelley witnesses
- [x] Multiple-output external tx (for cardano-transactions construction) for Shelley witnesses
- [x] Single-output external tx (for cardano-transactions construction) for Byron witnesses
- [x] Point to proper commit when https://github.com/input-output-hk/cardano-transactions/pull/17 is merged

# Comments

### CODE REVIEW

File: lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Transactions.hs [hunk 1]
```diff
@@ -533,6 +551,190 @@ spec = do
...
+        verify r6
+            [ expectSuccess
+            , expectResponseCode HTTP.status202
+            , expectField (#amount . #getQuantity) $
+                between (feeMin1 + amt1, feeMax1 + amt1)
+            , expectField (#direction . #getApiT) (`shouldBe` Outgoing)
+            , expectField (#status . #getApiT) (`shouldBe` Pending)
+            ]
+--}
+{--
```
Comment (paweljakubas) [inline lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Transactions.hs]: this should be uncommented and should pass. now I have : 
```
      uncaught exception: IOException of type UserError
       user error (expected a successful response but got an error: DecodeFailure "Error in $: parsing Cardano.Wallet.Api.Types.ApiWallet(ApiWallet) failed, key \"id\" not found: Response {responseStatus = Status {statusCode = 409, statusMessage = \"Conflict\"}, responseVersion = HTTP/1.1, responseHeaders = [(\"Transfer-Encoding\",\"chunked\"),(\"Date\",\"Sat, 15 Aug 2020 18:40:10 GMT\"),(\"Server\",\"Warp/3.3.5\")], responseBody = \"{\\\"code\\\":\\\"wallet_already_exists\\\",\\\"message\\\":\\\"This operation would yield a wallet with the following id: b062e8ccf3685549b6c489a4e94966bc4695b75b However, I already know of a wallet with this id.\\\"}\", responseCookieJar = CJ {expose = []}, responseClose' = ResponseClose}")
```
strange, but maybe something more needs to be changed to accommodate external tx

File: stack.yaml [hunk 2]
```diff
@@ -9,6 +9,7 @@ packages:
 - lib/test-utils
 - lib/jormungandr
 - lib/shelley
+- ../cardano-transactions
```
Comment (paweljakubas) [inline stack.yaml]: this is temporary, I just use this branch https://github.com/input-output-hk/cardano-transactions/pull/17. When this PR lands into master, we need to remove that change before merging this PR. but revert below change and bump commit

File: lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Transactions.hs (hunk 1 above)
Comment (paweljakubas) [reply lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Transactions.hs]: this is from line 717

Comment (paweljakubas) [reply lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Transactions.hs]: now it works

File: lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Transactions.hs [hunk 3]
```diff
@@ -533,6 +549,178 @@ spec = do
...
+                "passphrase": #{fixturePassphrase}
+                } |]
+        r1 <- request @ApiWallet ctx (Link.postWallet @'Shelley) Default walletPostData
+        verify r1
+            [ expectSuccess
+            , expectResponseCode HTTP.status201
+            , expectField
+                (#balance . #getApiT . #available) (`shouldBe` Quantity 0)
+            ]
+        let (_, Right wSrc) = r1
```
Comment (KtorZ) [inline lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Transactions.hs]: use instead: `getFromResponse id r1`

File: lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Transactions.hs [hunk 4]
```diff
@@ -533,6 +549,178 @@ spec = do
...
+            (Link.createTransaction @'Shelley wFaucet) Default payload1
+
+        verify r2
+            [ expectSuccess
+            , expectResponseCode HTTP.status202
+            , expectField (#amount . #getQuantity) $
+                between (feeMin + amt, feeMax + amt)
+            , expectField (#direction . #getApiT) (`shouldBe` Outgoing)
+            , expectField (#status . #getApiT) (`shouldBe` Pending)
+            ]
```
Comment (KtorZ) [inline lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Transactions.hs]: What exactly is this for :thinking: ? How does it relate to the testing of external tx?

**Comment by paweljakubas** [comment]
bors try

File: stack.yaml [hunk 5]
```diff
@@ -54,7 +54,7 @@ extra-deps:
 # Not strictly a dependency at present, but building it here to get
 # access to the cardano-tx cli.
```
Comment (rvl) [inline stack.yaml]: This comment no longer applies, so should be removed.

**Review by rvl — APPROVED** [review]
Looks good. The setup of test data looks quite tricky. Perhaps this could be scripted outside of the test cases?

File: stack.yaml (hunk 5 above)
Comment (paweljakubas) [reply stack.yaml]: cleaned in https://github.com/input-output-hk/cardano-wallet/pull/2043/commits/ae1f415d5a873eb3a50e501452e698ef6b6c71b9

**Pull request merged**